pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -0,0 +1,769 @@
+            self.authentication_method = AuthenticationMethod.ORG_KEY
+            self._organization_entry.set_text(self.subscription_request.organization)
+            set_in_entry = bool(self._activation_key_entry.get_text())
+            set_in_module = self.subscription_request.activation_keys.type == SECRET_TYPE_HIDDEN
+            if set_in_module:
+                if not set_in_entry:
+                    self.enable_activation_key_placeholder(True)
+            else:
+                self._activation_key_entry.set_text("")
+                self.enable_activation_key_placeholder(False)
+
+        # check if system purpose part of the spoke should be visible
+        self.system_purpose_visible = self.system_purpose_data.check_data_available()
+
+        # NOTE: the fill_combobox() function makes sure to remove old data from the
+        #       combo box before filling it
+
+        # role
+        fill_combobox(self._system_purpose_role_combobox,
+                      system_purpose_data.role,
+                      self._subscription_module.proxy.GetValidRoles())
+        # SLA
+        fill_combobox(self._system_purpose_sla_combobox,
+                      system_purpose_data.sla,
+                      self._subscription_module.proxy.GetValidSLAs())
+        # usage
+        fill_combobox(self._system_purpose_usage_combobox,
+                      system_purpose_data.usage,
+                      self._subscription_module.GetValidUsageTypes())
+
+        # Insights
+        self._insights_checkbox.set_active(self._subscription_module.InsightsEnabled)
+
+        # set HTTP proxy data
+        proxy_hostname = self.subscription_request.server_proxy_hostname
+        proxy_port = self.subscription_request.server_proxy_port
+        proxy_port_set = proxy_port >= 0
+        proxy_username = self.subscription_request.server_proxy_user
+        proxy_password_secret = self.subscription_request.server_proxy_password
+        proxy_password_set = proxy_password_secret.type == SECRET_TYPE_HIDDEN
+        self.http_proxy_visible = proxy_hostname or proxy_username or proxy_password_set
+        if proxy_hostname:
+            proxy_url = proxy_hostname
+            if proxy_port_set:
+                proxy_url = "{}:{}".format(proxy_url, proxy_port)
+            self._http_proxy_location_entry.set_text(proxy_url)
+        # HTTP proxy username
+        self._http_proxy_username_entry.set_text(proxy_username)
+        # HTTP proxy password
+        set_in_entry = bool(self._http_proxy_password_entry.get_text())
+        secret_type = self.subscription_request.server_proxy_password.type
+        set_in_module = secret_type == SECRET_TYPE_HIDDEN
+        if set_in_module:
+            if not set_in_entry:
+                self.enable_http_proxy_password_placeholder(True)
+        else:
+            self._http_proxy_password_entry.set_text("")
+            self.enable_http_proxy_password_placeholder(False)
```

Comment:
Second part which can be moved to a private method.

Reply:
Moved.